File: battery_main.jl
include("battery_ecm.jl")

function main()
    ah = 3.0
    soc = 1.0
    current = -3.0
    dt = 1.0
    v1 = 0.0
    time_end = 3600.0
    loop_num = Int(time_end / dt)

    time_ = 0.0
    ccv = fn_soc_ocv(soc)

    time_record_list = [0.0]
    ccv_record_list = [ccv]
    soc_record_list = [soc]

    for i in 1:loop_num
        soc, ccv, hgen, v1 = equivalent_circuit_model(soc, current, dt, v1, ah)
        time_ += dt
        push!(time_record_list, time_)
        push!(ccv_record_list, ccv)
        push!(soc_record_list, soc)
    end

    open("result.csv", "w") do fout
        for i = 1:length(time_record_list)
            println(fout, "$(time_record_list[i]),$(soc_record_list[i]),$(ccv_record_list[i])")
        end
    end
end

main()

File: battery_main.js
const fs = require('fs');
const { equivalent_circuit_model, fn_soc_ocv } = require('./battery_ecm');

const ah = 3;
let soc = 1;
let current = -3;
let dt = 1;
let v1 = 0;
const time_end = 3600;
let loop_num = Math.floor(time_end / dt);

let time_ = 0;
let ccv = fn_soc_ocv(soc);

let time_record_list = [0];
let ccv_record_list = [ccv];
let soc_record_list = [soc];

for (let i = 0; i < loop_num; i++) {
    let result = equivalent_circuit_model(soc, current, dt, v1, ah);
    soc = result.soc;
    ccv = result.ccv;
    hgen = result.hgen;
    v1 = result.v1;
    time_ += dt;
    time_record_list.push(time_);
    ccv_record_list.push(ccv);
    soc_record_list.push(soc);
}

// Prepare data for CSV
let csvContent = "Time,CCV,SOC\n";
for (let i = 0; i < time_record_list.length; i++) {
    csvContent += `${time_record_list[i]},${ccv_record_list[i]},${soc_record_list[i]}\n`;
}

// Write to CSV file
fs.writeFile("output.csv", csvContent, (err) => {
    if (err) {
        console.error(err);
    } else {
        console.log("CSV file has been saved!");
    }
});
